domain/learningandworkprogress/src/main/kotlin/uk/gov/justice/digital/hmpps/domain/learningandworkprogress/review/SentenceType.kt

```diff
@@ -12,3 +15,17 @@ enum class SentenceType {
   UNKNOWN,
   OTHER,
 }
+
+/**
+ * Return the prisoner's effective sentence type.
+ * There is an order or precedence:
+ *   * If they have the `isIndeterminate` flag set, they are considered INDETERMINATE_SENTENCE, regardless of anything else
+ *   * If they have the `isRecall` flag set, they are considered RECALL
+ *   * Else return the sentence type from the prisoner record
+ */
+fun effectiveSentenceType(prisonerSentenceType: SentenceType, prisonerHasIndeterminateFlag: Boolean, prisonerHasRecallFlag: Boolean) =
```

Comment:
Function moved from `ReviewService` so it can be used by both `ReviewService` and `ReviewScheduleService`
(and it feels like it lives here because it's related to SentenceType 👍 )